docs: Update database-types (#5884)
ddev/ddev: docs/content/users/extend/database-types.md

DDEV supports many versions of the MariaDB, MySQL, and PostgreSQL database servers.

> The following database types are supported out of the box:

Reviewer: Is "out of the box" too much of a metaphor? Should we use something less metaphoric?

- MariaDB 5.5-10.8 and 10.11
- MySQL 5.5-8.0
- Postgres 9-16

> The default database type is MariaDB, and the default version is currently 10.4.

Reviewer: ```suggestion
The default database type is MariaDB, and the default version is currently 10.11.
```

> New LTS versions of each of these are typically added soon after release. The very old versions are kept for compatibility with older projects.

Reviewer: ```suggestion
New LTS versions of each of these are typically added soon after they are released. The very old versions are kept for compatibility with older projects.
```

> (New LTS versions of each of these are typically added soon after release. The very old versions are kept for compatibility with older projects.)

Reviewer 2: ```suggestion
New LTS versions of each of these are typically added soon after release. The very old versions are kept for compatibility with older projects.
```
I think it's better without parentheses when laid out like this.

Author: Agreed. I had it without, then worried about what Matt would say. :)

- If you change the database type or version in an existing project, the existing database will not be compatible with your change, so you’ll want to use [`ddev export-db`](../usage/commands.md#export-db) to save a dump first.
- When you change database type, destroy the existing database using [`ddev delete --omit-snapshot`](../usage/commands.md#delete) before changing, then after [`ddev start`](../usage/commands.md#start) use [`ddev import-db`](../usage/commands.md#import-db) to import the dump you saved.
- Snapshots are always per database type and database version. So if you have snapshots from MariaDB 10.2 and you switch to MariaDB 10.5, don’t expect to be able to restore the old snapshot.

## Other databases types

Through add-ons, additional database types are also supported.
The following official add-ons are available:

> - [ddev/ddev-mongo](https://github.com/ddev/ddev-mongo)

Reviewer: ```suggestion
- [`ddev/ddev-mongo`](https://github.com/ddev/ddev-mongo)
```

> - [ddev/ddev-sqlsrv](https://github.com/ddev/ddev-sqlsrv)

Reviewer: ```suggestion
- [`ddev/ddev-sqlsrv`](https://github.com/ddev/ddev-sqlsrv)
```
